src/main/java/one/util/streamex/StreamExInternals.java

```diff
@@ -963,6 +963,42 @@ S doClone() {
         }
     }
 
+
+    static abstract class IntCloneableSpliterator<S extends IntCloneableSpliterator<?>> implements Spliterator.OfInt,

```

Comment:
For me it seems that these three new classes are unnecessary. Looks like `CloneableSpliterator` can do the thing. No?